File: backend/app/src/main/java/com/lemon/DTO/TransactionResponse.java
package com.lemon.DTO;

import com.lemon.models.TransactionDirection;
import com.lemon.models.TransactionStatus;
import com.lemon.models.TransactionType;

import java.math.BigDecimal;


public record TransactionResponse(
        String id,
        String reference,
        TransactionType type,
        TransactionStatus status,
        BigDecimal amount,
        String debitPhone,
        String creditPhone,
        String description,
        String createdAt,

        TransactionDirection direction) {}

File: app/src/main/java/com/example/lemon/network/models/TransactionResponse.kt
package com.example.lemon.network.models

import com.example.lemon.model.TransactionStatus
import com.example.lemon.model.TransactionType
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.math.BigDecimal

@Serializable
data class TransactionResponse(

    val id: String,

    val reference: String,

    val type: TransactionType,

    val status: TransactionStatus,

    @Contextual
    val amount: BigDecimal,

    val debitPhone: String,

    val creditPhone: String,

    val description: String,
    val createdAt: String
)